tao_compiler/mlir/disc/transforms/tests/disc-shape-optimization.mlir

```diff
@@ -165,6 +165,47 @@ func @main(%arg0: tensor<?x?x?xf32>, %arg1: tensor<?x?xf32>) -> tensor<?x?xf32>
 
 // -----
 
+// Test arith.cmpi + arith::muli + arith::add
+// CHECK-LABEL: main
+func @main(%arg0: tensor<?x?x?xf32>) -> (index, index) {
```

Comment:
Nit: split the test into simple and tiny cases, one unit test one thing

Reply:
Actually, I did this by design. It was designed to check the chain of propagation is ok.